test/api/repository/survey_repository_test.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:flutter_config/flutter_config.dart';
+import 'package:survey_flutter_ic/api/exception/network_exceptions.dart';
+import 'package:survey_flutter_ic/api/repository/survey_repository.dart';
+import 'package:survey_flutter_ic/api/response/surveys_response.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:survey_flutter_ic/model/survey_model.dart';
+import '../../mocks/generate_mocks.mocks.dart';
+import '../../utils/file_utils.dart';
+
+void main() {
+  FlutterConfig.loadValueForTesting({
+    'CLIENT_ID': 'CLIENT_ID',
+    'CLIENT_SECRET': 'CLIENT_SECRET',
+  });
+
+  group('SurveyRepository', () {
+    late MockSurveyService mockSurveyService;
+    late SurveyRepository repository;
+
+    setUp(() {
+      mockSurveyService = MockSurveyService();
+      repository = SurveyRepositoryImpl(mockSurveyService);
+    });
+
+    test(
+        'When calling GetSurveys successfully, it emits the corresponding AuthModel',
+        () async {
+      final json = await FileUtils.loadFile('mock_response/surveys.json');
+      final expected = SurveysResponse.fromJson(json);
```

Comment:
Loading response from json file seems relevant to the integration test, for the unit test, we could create a SurveyResponse model for the service returns.

Reply:
@hoangnguyen92dn We will definitely use it in the `integration test`. As we have it, Can we apply them to use on both `Integration test` and `UnitTest`? So we don't create a model in case there're many fields. It will save time I think or What do you think? 🙏

Reply:
@Wadeewee It's fine then. I think it also makes sense to reflect the same json resource for both integration and unit test